### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kovpaka**

### CODE REVIEW

File: BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System.Threading.Tasks;
+using BeerCatalog.BL.Models;
+using BeerCatalog.BL.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BeerCatalog.Web.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class BeerCatalogController : Controller
```
Comment (siarheikarabko) [inline BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs:11]: ApiController should inherit ControllerBase class.
https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-5.0#controllerbase-class

File: BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public BeerCatalogController(IBeerService service)
+        {
+            this.service = service;
+        }
+
+        [HttpGet("Beers/{id}")]
+        public async Task<IActionResult> GetBeer(int id)
+        {
+            Beer beer = await service.GetBeerById(id);
+            return Json(beer);
```
Comment (siarheikarabko) [inline BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs:24]: You could just return beer and do not wrap it in Json()
https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-5.0#controllerbase-class

File: BeerCatalog/BeerCatalog.Client/Repositories/PunkApiClient.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+            return await GetData<Beer>(currentUrl);
+       }
+
+        private async Task<IReadOnlyList<T>> GetData<T>(string url)
+        {
+            IReadOnlyList<T> result;
+            try
+            {
+                using HttpResponseMessage responseMessage = await client.GetAsync(url);
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply BeerCatalog/BeerCatalog.Client/Repositories/PunkApiClient.cs]: Wow, it's cool, thank you, fixed

File: BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs (hunk 1 above)
Comment (kovpaka) [reply BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs]: Got it

File: BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs (hunk 2 above)
Comment (kovpaka) [reply BeerCatalog/BeerCatalog.Web/Controllers/BeerCatalogController.cs]: fixed

File: BeerCatalog/BeerCatalog.Client/ConfigServicesCollectionExtension.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using PunkApi.Client.Options;
+using PunkApi.Client.Repositories;
+using PunkApi.Client.Repositories.Interfaces;
+
+namespace PunkApi.Client
+{
+    public static class ConfigServicesCollectionExtension
+    {
+        public static IServiceCollection AddClient(
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply BeerCatalog/BeerCatalog.Client/ConfigServicesCollectionExtension.cs]: Fixed

File: BeerCatalog/BeerCatalog.BL/Services/BeerService.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        public async Task<Beer> GetBeerById(int id)
+        {
+            Beer beer = beerMapper.Map(await punkApiClient.GetBeerById(id));
+
+            return beer;
+        }
+
+        public async Task<IReadOnlyList<Beer>> GetBeers(string name, PageOptions pageOptions)
+        {
+            IReadOnlyList<Beer> beers = beerMapper.Map(await punkApiClient.GetBeers(name, pageOptionsMapper.Map(pageOptions)));
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply BeerCatalog/BeerCatalog.BL/Services/BeerService.cs]: Ok

File: BeerCatalog/BeerCatalog.Client/Repositories/PunkApiClient.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+       public async Task<IReadOnlyList<Beer>> GetBeers(string name, PageOptions pageOptions)
+       {
+           string currentUrl = $"{urlOptions.Url}" +
+                               $"?page={pageOptions.PageNumber}&per_page={pageOptions.PageSize}" +
+                               $"{(name != null ? $"&beer_name={name}" : "")}";
+
+            return await GetData<Beer>(currentUrl);
+       }
+
+        private async Task<IReadOnlyList<T>> GetData<T>(string url)
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply BeerCatalog/BeerCatalog.Client/Repositories/PunkApiClient.cs]: Fixed

**Pull request merged**